### PR TITLE
Property test the SSEClient

### PR DESCRIPTION
**Pull request by nietaki**
I also added `test.watch` and started using a test support directory, as most projects I saw did instead of requiring source files from the test files.

I was hoping to find some bugs in the implementation, but it seems to be holding up.

I don't have strong opinions about where the new tests should live, maybe they should be co-located with some of the other ones, feel free to change.

### CODE REVIEW

**Comment by nietaki** [comment]
btw, I didn't import `StreamData` in the test file to learn what is where in the project when I was implementing it, there's no reason not to import it and use the functions directly.

File: test/server_sent_event/client_property_test.exs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    StreamData.frequency([
+      {9, simple_sse()},
+      {1, comment_sse()}
+    ])
+  end
+
+  def ascii_string_with_newlines do
+    StreamData.string(:ascii)
+    |> StreamData.list_of()
+    |> StreamData.map(&Enum.join(&1, "\n"))
```
Comment (CrowdHailer) [inline test/server_sent_event/client_property_test.exs]: Is it worth sometimes putting `\r` or `\r\n` in as the newline

File: test/server_sent_event/client_property_test.exs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      packets =
+        split_pairs
+        |> Enum.map(fn [from, to] -> binary_part(whole_response, from, to - from) end)
+
+      assert Enum.join(packets, "") == whole_response
+
+      Enum.each(packets, fn packet ->
+        :ok = :gen_tcp.send(socket, packet)
+        # try to make sure erlang's magic is not going to merge the packets
+        Process.sleep(1)
```
Comment (CrowdHailer) [inline test/server_sent_event/client_property_test.exs]: I think you might need more than one ms of sleep.

File: test/server_sent_event/client_property_test.exs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      # preparing the connection
+      {port, listen_socket} = listen()
+      {:ok, client} = AutoConnect.start_link(port)
+
+      {:ok, socket} = accept(listen_socket)
+      {:ok, first_request} = :gen_tcp.recv(socket, 0, 1_000)
+      assert String.contains?(first_request, "accept: text/event-stream")
+      assert String.contains?(first_request, "\r\n\r\n")
+
+      :ok = :gen_tcp.send(socket, [@first_response])
```
Comment (CrowdHailer) [inline test/server_sent_event/client_property_test.exs:33]: Not necessarily part of the problem but I would be tempted to add first response to the whole_response string so this test includes splitting the response head

Comment (nietaki) [reply test/server_sent_event/client_property_test.exs:33]: yeah, I was thinking the same thing.

I started doing it, but I realised the client assumes the headers are going to arrive in one packet. 

I'll add it as an issue.

File: test/server_sent_event/client_property_test.exs [hunk 4]
```diff
@@ -124,9 +124,17 @@ defmodule ServerSentEvent.ClientPropertyTest do
...
-    StreamData.string(:ascii)
-    |> StreamData.list_of()
-    |> StreamData.map(&Enum.join(&1, "\n"))
+    lines =
+      StreamData.string(:ascii)
+      |> StreamData.list_of()
+
+    new_line_string = StreamData.member_of(["\n", "\r\n", "\r"])
+
+    {lines, new_line_string}
```
Comment (CrowdHailer) [inline test/server_sent_event/client_property_test.exs:133]: Well this is a neat little solution

**Pull request merged**